### DISCUSSION TITLE
How to implement multiple custom schema directives (visitSchemaDirectives) through utils package for v8 ?

### DISCUSSION THREAD

**Discussion by SidKhanna296** [apollographql/apollo-server#6128] [Q&A] [answered]
**_Note: I have asked this question in graphql-tools repo too just seeing if someone here has encountered the same issue as help would be useful and it is ancillary behaviour_**

I am migrating graphql tools from v7 to v8 and need to update my custom schema directive.
I have a subgraph which uses multiple schema directive to implement different functionality (customDirective1, customDirective2 ...)

in my v7 implementation, I had:

```
const schema = buildSubgraphSchema([
        {
            typeDefs: getSchema(),
            resolvers,
        },
    ]);

    SchemaDirectiveVisitor.visitSchemaDirectives(schema, {
        directive1Name: customDirective1Func,
        directive2Name: customDirective2Func,
        ...
    });

const server = new ApolloServer({schema})
```
in the apollo and graphql/tools docs the example only highlights how to implement single custom directives for v8:

```
let schema = makeExecutableSchema({
  typeDefs,
  resolvers
});

// Transform the schema by applying directive logic
schema = schemaDirective1Func(schema, 'directive1Name');
```

My question becomes if I had multiple directives, what is considered the standard on implementing it or how would you implement it?

I could add another line like this, but the below seems clunky
`schema = schemaDirective2Func(schema, 'directive2Name');`

Another thought I had was using a reducer
`[schemaDirective1Func(directive1Name), schemaDirective2Func(directive2Name)].reduce( (previousSchema, directive) => directive(previousSchema), schema)`

Or is the recommended pattern to just have all the transformers and typedefs mapped to a standard field and just call that ?
i.e.:
```
const { schemaDirectiveTransformer: schemaDirective1FuncTransformer } = schemaDirective1Func('directive1Name')
const { schemaDirectiveTransformer: schemaDirective2FuncTransformer } = schemaDirective2Func('directive2Name')

schema = schemaDirectiveTransformer(schema);
```

But was not sure if any of this is a recommended standard, if this would work or if there is a function that handles this similar logic like visitSchemaDirectives ?

**Comment by glasser** [comment]
I do think this discussion is best had over on `graphql-tools` because this really is about using their API. It does seem like these patterns would all work fine?

**Comment by SidKhanna296** [comment, marked as answer]
thanks - yeah this question was definitely meant for graphql-tools but was curious anyone else encountered it here - 
anyways I got my answer guess I was on the right track in terms of using reducer - https://github.com/ardatan/graphql-tools/discussions/3419
